app/static/js/base.js

```diff
@@ -1,16 +1,47 @@
+const flashMessageResponse = function flashEventResponse(message){
+  if (message.slice(-8) == "deleted."){
+
+    return `<strong><a href="/event/undo" style="color: dark-green;" id="restoreDeleted">Undo</a></strong>` 
+  }
+  return '';
+}
+
+function undoGeneralEvent(){
+  undoEvent = {from: 'ajax'}
+  $.ajax({
+    url: "/event/undo",
+    type: "POST",
+    data: undoEvent,
+    success: function(){
+      alert("Successfully undone!")
+    },
+    error: function(error, status){
+      console.log(error, status)
+    }
+  })
+}
+
 function msgFlash(flash_message, status){
     if (!["success", "warning", "info", "danger"].includes(status)) status = "danger";
     $("#flash_container").prepend(`
-      <div class="alert alert-${status} alert-dismissible" role="alert">${flash_message}
-        <button type="button" class="btn-close kiosk-hide" data-bs-dismiss="alert" aria-label="Close"></button>
+      <div class="alert alert-${status} alert-dismissible alert-success" role="alert">${flash_message}
+        ${flashMessageResponse(flash_message)}
+        <button type="button" class="btn-close kiosk-hide"  id="flashResponded"  aria-label="Close"></button>
       </div>`);
-    $(".alert").delay(5000).fadeOut();
+    $("#flashResponded").click(function(){
+      $(".alert").delay(1000).fadeOut();
+    })
 }
+
+
+
 $(document).ready(function() {
     $("select[name='newuser']").on('change', function(e) {
         $(e.target).parent().submit();
     });
-
+    $('#restoredDeleted').click(function(){
+      undoGeneralEvent($('#restoredDeleted').val())
```

Comment:
It doesn't look like we use this value in the function. Either we need to add an argument to the function and use it, or we can remove it here.

Reply:
We actually don't need this click handler anymore